management/distance-target.yaml

```diff
@@ -0,0 +1,261 @@
+openapi: 3.0.1
+info:
+  title: Adzerk Management API - Distance Targeting
+  description: Advertiser related Adzerk Management API
+  version: '1.0'
+servers:
+  - url: 'https://api.adzerk.net'
+tags:
+  - name: distance-target
+    description: Manage Distance Targeting
+paths:
+  '/v1/flight/{flightId}/distance':
+    post:
+      tags:
+        - distance-targeting
+      operationId: create
+      security:
+        - ApiKeyAuth: []
+      parameters:
+        - name: flightId
+          description: The Flight Id to associate with the Distance Targeting geometry object
+          in: path
+          required: true
+          schema:
+            type: integer
+            format: int32
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                Distance:
+                  type: integer
+                  format: float
+                  nullable: true
+                Latitude:
+                  type: integer
+                  format: float
+                  nullable: true
+                Longitude:
+                  type: integer
+                  format: float
+                  nullable: true
+                StreetAddress:
+                  type: string
+                  nullable: true
+      responses:
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/distance-target.yaml#/schemas/BadRequest'
+        503:
+          description: Geocoding service unavanilable
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/distance-target.yaml#/schemas/ServiceUnavailable'
+        200:
+          description: The newly created Distance Target
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/distance-target.yaml#/schemas/DistanceTarget'
+  '/v1/flight/{flightId}/distance/batch':
+    post:
+      tags:
+        - distance-targeting
+      operationId: batchUpload
+      security:
+        - ApiKeyAuth: []
+      parameters:
+        - name: flightId
+          description: The Flight Id to associate with the Distance Targeting geometry object
+          in: path
+          required: true
+          schema:
+            type: integer
+            format: int32
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                ReplaceExisting:
+                  type: boolean
+                  nullable: true
+                Geometries:
+                  type: array
+                  items: 
+                    $ref: './schemas/distance-target.yaml#/schemas/BatchUpload'
```

Comment:
Because of how the JS SDK parses these, I tend not to use refs inside of the body schema. Especially when they're not re-used.

Reply:
Updated